### PR TITLE
DLS-9782: added session cache and modes, and added tests

### PR DESCRIPTION
**Pull request by JayenKerai**

### CODE REVIEW

File: it/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersUnincorporatedBodyNameControllerISpec.scala [hunk 1]
```diff
@@ -1,90 +1,138 @@
...
+          res.status mustBe 200
+          val page = Jsoup.parse(res.body)
+          page.title() must include("What is the unincorporated body name?")
+          page.getElementsByTag("h1").text() must include("What is the unincorporated body name?")
+          val companyName = page.getElementById("unincorporatedBodyName_value")
+          companyName.attr("value") mustBe "company name"
+        }
+      }
+
+      "redirect the user to the start of the BusinessPartners journey" when {
```
Comment (BackRowOla) [inline it/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersUnincorporatedBodyNameControllerISpec.scala:61]: The user is redirected to the start of the FHDDS journey now not the businessPartners journey

**Review by BackRowOla — CHANGES_REQUESTED** [review]
Looks good for the most part, just a few minor changes

File: app/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersController.scala [hunk 2]
```diff
@@ -63,7 +63,7 @@ class BusinessPartnersController @Inject()(
       businessType => {
         val nextUrl = businessType match {
           case BusinessPartnerType.UnincorporatedBody =>
-            routes.BusinessPartnersUnincorporatedBodyNameController.load()
+            routes.BusinessPartnersUnincorporatedBodyNameController.load(1, NormalMode)
```
[the remark this answers is not in the document]
Comment (JayenKerai) [reply app/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersController.scala]: Changes made

File: app/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersUnincorporatedBodyNameController.scala [hunk 3]
```diff
@@ -16,64 +16,73 @@
...
 
 import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
 
 class BusinessPartnersUnincorporatedBodyNameController @Inject()(
   ds: CommonPlayDependencies,
   view: Views,
   actions: Actions,
-  config: FrontendAppConfig)(
+  config: FrontendAppConfig,
```
[the remark this answers is not in the document]
Comment (JayenKerai) [reply app/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersUnincorporatedBodyNameController.scala]: Changes made

File: it/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersUnincorporatedBodyNameControllerISpec.scala (hunk 1 above)
Comment (JayenKerai) [reply it/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersUnincorporatedBodyNameControllerISpec.scala]: Changes made

File: app/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersUnincorporatedBodyNameController.scala [hunk 4]
```diff
@@ -16,64 +16,73 @@
...
+  val sessionCache: SessionRepository)(
   cc: MessagesControllerComponents
-) extends AppController(ds, cc) {
+)(implicit val ec: ExecutionContext)
+    extends AppController(ds, cc) with ControllerHelper {
 
   import actions._
 
   val backUrl: String = routes.BusinessPartnersController.load(1, NormalMode).url
```
[the remark this answers is not in the document]
Comment (JayenKerai) [reply app/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersUnincorporatedBodyNameController.scala]: Changes made

**Review by BackRowOla — APPROVED** [review]
LGTM 👍

**Pull request merged**